bin/docker-image-tool.sh

```diff
@@ -41,6 +41,18 @@ function image_ref {
   echo "$image"
 }
 
+function docker_push {
+  local image_name="$1"
+  if [ ! -z $(docker images -q "$(image_ref ${image_name})") ]; then
+    docker push "$(image_ref ${image_name})"
+    if [ $? -ne 0 ]; then
+      error "Failed to push $image_name Docker image."
+    fi
+  else
+    echo "$(image_ref ${image_name}) image not found. Skipping push for this image."
```

Comment:
Currently the script could incorrectly go down this path if an image build fails because it doesn't exit in that case

Reply:
We do fail with non zero exit code when build or push fails. We are skipping push only in case or images not found.